min_stack.py

```diff
@@ -0,0 +1,73 @@
+# Approach: Implemented the min stack with the data structure 'array' similar to precourse 1 problem (Implement stack using array)
+# Choice: When we implemented stack using array, the methods push, pop, peek etc all ran in constant space and time
+# We declare a minStack which holds the min value at every instant 
+# since we need to keep track of this minimum value everytime we push a new element or pop off an element, the min value will keep changing
+# Time Complexity : O(1) # append, pop peek all are constant time operations
+# Space Complexity : O(1) ## append, pop peek all are constant space operations
+# Succesfully ran in leetcode
+# Base Case: Not considering an empty stack as constraint mentions non-empty stack
+
+# Doubt: I have declared a minStack [] but that only holds the min value appended for every push method. 
+# I mentioned O(1), since if methods push/pop are run independantly it is hardly storing one value(append) or popping one val at a time, which is constant space/time
+# Should I mention this as the case or the overall when we need to perform on 'n' such elements in the stack? Not quite sure
+
+class MinStack:
```

Comment:
Please error guard pop(), top() & getMin() methods to prevent from operating on an empty stack, the program will crash otherwise.

Reply:
Thanks for the feedback, i referred to the constraint on leetcode and it said `Methods pop, top and getMin operations will always be called on non-empty stacks.` 

In this case should I still address the constraint?

Reply:
Yes, it is always good to add on edge cases that Leetcode might or might not suggest, ultimately you will not be coding on a platform like leetcode for your interviews. For any further doubts please message me on Slack.